worker/src/lib.rs

```diff
@@ -135,6 +134,93 @@ pub async fn witness_tweet(
         };
         vc.evidence = Some(OneOrMany::One(evidence));
 
+        info!("{:?}", vc);
+
+        let proof = jserr!(
+            vc.generate_proof(
+                &sk,
+                &LinkedDataProofOptions {
+                    verification_method: Some(format!("{}#controller", SPRUCE_DIDWEB)),
+                    ..Default::default()
+                }
+            )
+            .await
+        );
+        vc.proof = Some(OneOrMany::One(proof));
+        Ok(jserr!(serde_json::to_string(&vc)).into())
+    })
+}
+
+#[wasm_bindgen]
+pub async fn witness_discord(
+    secret_key_jwk: String,
+    public_key_tezos: String,
+    discord_authorization_key: String,
+    discord_handle: String,
+    channel_id: String,
+    message_id: String,
+) -> Promise {
+    use log::Level;
+    console_log::init_with_level(Level::Trace).expect("error initializing log");
+
+    future_to_promise(async move {
+        let pk: JWK = jserr!(jwk_from_tezos_key(&public_key_tezos));
+        let sk: JWK = jserr!(serde_json::from_str(&secret_key_jwk));
+        let discord_res = jserr!(
+            discord::retrieve_discord_message(discord_authorization_key, channel_id, message_id)
+                .await
+        );
+        let formatted_discord_handle = format!(
+            "{}#{}",
+            discord_res.author.username, discord_res.author.discriminator
+        );
+        let mut vc = jserr!(discord::build_discord_vc(&pk, &formatted_discord_handle));
+
+        // Check for matching handles
+        if discord_handle != formatted_discord_handle {
```

Comment:
If Discord handles aren't case sensitive, you should compare them in their lowercased form

Reply:
I believe they are case sensitive